awx/ui_next/src/screens/Credential/CredentialList/CredentialListItem.jsx

```diff
@@ -66,16 +72,33 @@ function CredentialListItem({
           id={labelId}
         >
           {canEdit ? (
-            <Tooltip content={i18n._(t`Edit Credential`)} position="top">
-              <Button
-                aria-label={i18n._(t`Edit Credential`)}
-                variant="plain"
-                component={Link}
-                to={`/credentials/${credential.id}/edit`}
-              >
-                <PencilAltIcon />
-              </Button>
-            </Tooltip>
+            <>
+              <Tooltip content={i18n._(t`Edit Credential`)} position="top">
+                <Button
+                  isDisabled={disableButtons}
+                  aria-label={i18n._(t`Edit Credential`)}
+                  variant="plain"
+                  component={Link}
+                  to={`/credentials/${credential.id}/edit`}
+                >
+                  <PencilAltIcon />
+                </Button>
+              </Tooltip>
+              {credential.summary_fields.user_capabilities.copy && (
+                <CopyButton
+                  isDisabled={disableButtons}
+                  css="grid-column: 3"
+                  itemName={credential.name}
+                  disableButtons={setDisableButtons}
+                  copyItem={async () => {
+                    await CredentialsAPI.copyCredential(credential.id, {
+                      name: `${credential.name}@${timeOfDay()}`,
+                    });
+                    await fetchCredentials();
+                  }}
```

Comment:
Use `useCallback` here (see comments in CopyButton above). You want this function to be protected by a corresponding dependency array with `fetchCredentials` etc in it.